### PR TITLE
[WIP] Try out sergei-lapin/napt

### PR DESCRIPTION
**Pull request by chrisbanes**
Migrates to @sergei-lapin's [napt](https://github.com/sergei-lapin/napt), to replace kapt.

- Had to disable Hilt's aggregating task as it doesn't seem to work with napt
- gitignore NaptTrigger

### CODE REVIEW

**Comment by sergei-lapin** [comment]
Hi, @chrisbanes ! Right now, I think, you can get this working via disabling Hilt's aggregating task:
```
hilt {
    enableAggregatingTask = false
}
```
while I'm thinking of a better way to avoid this kind of errors

**Comment by chrisbanes** [comment]
> Hi, @chrisbanes ! Right now, I think, you can get this working via disabling Hilt's aggregating task:
> 
> ```
> hilt {
>     enableAggregatingTask = false
> }
> ```
> 
> while I'm thinking of a better way to avoid this kind of errors

Hi @sergei-lapin, I briefly tried that but it still didn't work. I'll try again though

File: build.gradle [hunk 1]
```diff
@@ -93,9 +93,10 @@ subprojects {
         }
     }
 
-    plugins.withId(rootProject.libs.plugins.hilt.get().pluginId) {
-        hilt.enableAggregatingTask = true
-    }
+    // Disable this whilst trying out napt #951
+    // plugins.withId(rootProject.libs.plugins.hilt.get().pluginId) {
+    //     hilt.enableAggregatingTask = true
```
Comment (sergei-lapin) [inline build.gradle:98]: it appears to be [enabled by default](https://github.com/google/dagger/blob/master/java/dagger/hilt/android/plugin/main/src/main/kotlin/dagger/hilt/android/plugin/HiltExtension.kt)

Comment (chrisbanes) [reply build.gradle:98]: Ah that must have changed 🤔

File: .gitignore [hunk 2]
```diff
@@ -38,3 +38,7 @@ org.eclipse.buildship.core.prefs
 .classpath
 .project
 bin/
+
+# napt triggers
+# https://github.com/sergei-lapin/napt
+NaptTrigger.java
```
Comment (Goooler) [inline .gitignore:44]: We can put generated `NaptTrigger.java` files into proper packages, and [disabling NaptTrigger generation](https://github.com/sergei-lapin/napt#disabling-napttrigger-generation).

Comment (chrisbanes) [reply .gitignore:44]: We can't disable the triggers as we don't have any Java sources. The triggers ensure that napt runs on Kotlin-only projects.

Comment (Goooler) [reply .gitignore:44]: I meant we can move the triggers into java sourceSets and disable `generateNaptTrigger`. 

image
image

Comment (chrisbanes) [reply .gitignore:44]: Yeah we could, but I quite like that they're not polluting the 'actual' package we use. As they're generated code, it doesn't really matter imo. WDYT?

Comment (Goooler) [reply .gitignore:44]: No problem.

**Pull request merged**